### ISSUE TITLE
extract Plugin: Can automatically create the decompression directory

### ISSUE THREAD

**Issue by rehack** [ohmyzsh/ohmyzsh#11537]
### If the feature request is for a plugin or theme, specify it here.

extract plugin

### If the feature solves a problem you have, specify it here.

If there is no package folder in the compressed file, a bunch of scattered files may appear after using extract plugin to extract.

### Describe the proposed feature.

For compressed files without folder package inside, can they be automatically put into a folder after decompression

**Comment by carlosala** [comment]
I think it's really interesting! Even though, seems difficult to have a standard way to do it with every kind of archive, but if you want to work with it, feel free to do it and I'd be glad to review it!

**Comment by rehack** [comment]
> I think it's really interesting! Even though, seems difficult to have a standard way to do it with every kind of archive, but if you want to work with it, feel free to do it and I'd be glad to review it!

That's exactly what I think. 
It takes a lot of work to determine the number of files inside every kind of archive, and below is just a simple example of a gz compressed file, which contains a lot of code.

```bash
#!/bin/bash

num_files=$(tar -tf file.tar.gz | wc -l)

if [ $num_files -gt 1 ]; then
  mkdir -p /path/to/folder/
  tar -xzvf file.tar.gz -C /path/to/folder/
else
  tar -xzvf file.tar.gz
fi
```

**Comment by mcornella** [comment]
A possible hack would be to always extract on a folder and then move it out if only a folder exists in it after extraction.

**Comment by carlosala** [comment]
That's a great option

**Comment by mcornella** [comment]
Implemented this in #11596. You can test it out with `omz pr test 11596`, let me know if you experience any issues.
